src/overlays/actors/ovl_Bg_Haka_Tubo/z_bg_haka_tubo.c

```diff
@@ -31,18 +30,199 @@ const ActorInit Bg_Haka_Tubo_InitVars = {
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Tubo/BgHakaTubo_Init.s")
+ColliderCylinderInit D_80881B60 = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+    { 25, 60, 30, { 0, 0, 0 } },
+};
+
+ColliderCylinderInit D_80881B8C = {
+    { COLTYPE_UNK10, 0x11, 0x00, 0x09, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x20000000, 0x01, 0x04 }, { 0x00000008, 0x00, 0x00 }, 0x19, 0x00, 0x01 },
+    { 60, 45, 235, { 0, 0, 0 } },
+};
+
+s32 D_80881BB8 = 0;
+
+InitChainEntry D_80881BBC[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+Vec3f D_80881BC0 = { 0.0f, 0.0f, 0.0f };
+
+extern CollisionHeader D_060108B8;
+extern Gfx D_0600FE40[];
+extern UNK_TYPE D_0400CD80;
+extern Gfx D_040184B0[];
+
+void BgHakaTubo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTubo* this = THIS;
+    s32 pad;
+    s32 sp24 = 0;
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80881BBC);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_UNK3);
+    DynaPolyInfo_Alloc(&D_060108B8, &sp24);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp24);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &D_80881B60);
+    Collider_InitCylinder(globalCtx, &this->colliderFlames);
+    Collider_SetCylinder(globalCtx, &this->colliderFlames, &this->dyna.actor, &D_80881B8C);
+    this->fireScroll = Math_Rand_ZeroOne() * 15.0f;
+    D_80881BB8 = 0;
+    this->actionFunc = BgHakaTubo_Idle;
+}
+
+void BgHakaTubo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTubo* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    Collider_DestroyCylinder(globalCtx, &this->colliderFlames);
+}
+
+void BgHakaTubo_Idle(BgHakaTubo* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+
+    if (this->dyna.actor.room == 12) { // 3 spinning pots room in Shadow Temple
+        this->dyna.actor.shape.rot.y += 0x180;
+        this->dyna.actor.posRot.pos.x = Math_Sins(this->dyna.actor.shape.rot.y - 0x4000) * 145.0f + -5559.0f;
+        this->dyna.actor.posRot.pos.z = Math_Coss(this->dyna.actor.shape.rot.y - 0x4000) * 145.0f + -1587.0f;
+    }
+    // Colliding with flame circle
+    if (this->colliderFlames.base.atFlags & 2) {
+        this->colliderFlames.base.atFlags &= ~2;
+        func_8002F71C(globalCtx, &this->dyna.actor, 5.0f, this->dyna.actor.yawTowardsLink, 5.0f);
+    }
+    // Colliding with hitbox inside the pot
+    if (this->collider.base.acFlags & 2) {
+        this->collider.base.acFlags &= ~2;
+        /*
+         * If the colliding actor is within a 50 unit radius and 50 unit height cylinder centered
+         * on the actor's position, break the pot
+         */
```

Comment:
Imo comments like that should still use `//` even when they end up being a couple lines. Otherwise they take unwanted space and stand out too much, while still being inline comments in intent.
```suggestion
         // If the colliding actor is within a 50 unit radius and 50 unit height cylinder centered
         // on the actor's position, break the pot
```